tests/integ/test_processing.py

```diff
@@ -695,3 +700,153 @@ def test_sklearn_with_network_config(sagemaker_session, sklearn_latest_version,
+        instance_count=1,
+        instance_type=cpu_instance_type,
+        volume_size_in_gb=100,
+        volume_kms_key=None,
+        output_kms_key=output_kms_key,
+        max_runtime_in_seconds=3600,
+        base_job_name="test-script-processor",
+        env={"DUMMY_ENVIRONMENT_VARIABLE": "dummy-value"},
+        tags=[{"Key": "dummy-tag", "Value": "dummy-tag-value"}],
+        sagemaker_session=sagemaker_session,
+    )
+
+    script_processor.run(
+        code=os.path.join(DATA_DIR, "dummy_script.py"),
+        inputs=[
+            ProcessingInput(
+                source=None,
+                destination=None,
+                input_name="input_dataset_definition",
+                dataset_definition=DatasetDefinition(
+                    local_path="/opt/ml/processing/input/script",
+                    redshift_dataset_definition=RedshiftDatasetDefinition(
+                        cluster_id="integ-test-cluster-prod-us-west-2",
+                        database="dev",
+                        db_user="awsuser",
+                        query_string="SELECT * FROM shoes",
+                        cluster_role_arn="arn:aws:iam::037210630505:role/RedshiftClusterRole-prod-us-west-2",
+                        output_s3_uri=f"s3://{sagemaker_session.default_bucket()}",
+                        output_format="CSV",
+                    ),
+                ),
+            )
+        ],
+        outputs=[
+            ProcessingOutput(
+                source="/opt/ml/processing/output/container/path/",
+                output_name="dummy_output",
+                s3_upload_mode="EndOfJob",
+            )
+        ],
+        arguments=["-v"],
+        wait=False,
+    )
+
+    job_description = script_processor.latest_job.describe()
+
+    dd_input = job_description["ProcessingInputs"][0]
+    assert dd_input["InputName"] == "input_dataset_definition"
+    assert dd_input["DatasetDefinition"] is not None
+    dd = dd_input["DatasetDefinition"]
+    assert dd["RedshiftDatasetDefinition"] is not None
+    assert "AthenaDatasetDefinition" not in dd
+    assert dd["DataDistributionType"] == "ShardedByS3Key"
+    assert dd["InputMode"] == "Pipe"
+    assert dd["LocalPath"] == "/opt/ml/processing/input/script"
+
+    assert dd["RedshiftDatasetDefinition"]["ClusterId"] == "integ-test-cluster-prod-us-west-2"
+    assert dd["RedshiftDatasetDefinition"]["Database"] == "dev"
+    assert dd["RedshiftDatasetDefinition"]["DbUser"] == "awsuser"
+    assert dd["RedshiftDatasetDefinition"]["QueryString"] == "SELECT * FROM shoes"
+    assert (
+        dd["RedshiftDatasetDefinition"]["ClusterRoleArn"]
+        == "arn:aws:iam::037210630505:role/RedshiftClusterRole-prod-us-west-2"
+    )
+    assert dd["RedshiftDatasetDefinition"]["OutputS3Uri"] == "s3://sagemaker-us-west-2-790336243319"
+    assert dd["RedshiftDatasetDefinition"]["OutputFormat"] == "CSV"
+    assert dd["RedshiftDatasetDefinition"]["OutputCompression"] == "GZIP"
```

Comment:
Looks like some of this can be refactored into a function.

Reply:
Done.